core/src/com/unciv/ui/components/Fonts.kt

```diff
@@ -273,10 +374,55 @@ object Fonts {
         }
     }
 
-    private val frameBuffer by lazy { FrameBuffer(Pixmap.Format.RGBA8888, Gdx.graphics.width, Gdx.graphics.height, false) }
+    private val frameBuffer by lazy {
+        // Size here is way too big, but it's hard to know in advance how big it needs to be.
+        // Gdx world coords, not pixels.
+        FrameBuffer(Pixmap.Format.RGBA8888, Gdx.graphics.width, Gdx.graphics.height, false)
+    }
     private val spriteBatch by lazy { SpriteBatch() }
+    private val transform = Matrix4()  // for repeated reuse without reallocation
 
+    /** Get a Pixmap for a "show ruleset icons as part of text" actor.
+     *
+     *  Draws onto an offscreen frame buffer and copies the pixels.
+     *  Caller becomes owner of the returned Pixmap and is responsible for disposing it.
+     *
+     *  Size is such that the actor's height is mapped to the font's ascent (close to
+     *  ORIGINAL_FONT_SIZE * GameSettings.fontSizeMultiplier), the actor is placed like a letter into
+     *  the total height as given by the font's metrics, and width scaled to maintain aspect ratio.
+     */
     fun getPixmapFromActor(actor: Actor): Pixmap {
+        // We want our - mostly circular - icon to match a typical large uppercase letter in height
+        // The drawing bounding box should have room, however for the font's leading and descent
+        val metrics = fontImplementation.getMetrics()
+        val boxHeight = ceil(metrics.height).toInt()
+        // Empiric slight size reduction - "correctly calculated" they just look a bit too big
+        val scaledActorHeight = metrics.ascent * 0.93f
+        val scaledActorWidth = scaledActorHeight * actor.width / actor.height
+        val boxWidth = ceil(scaledActorWidth).toInt()
+        // Nudge down by the border size if it's a Portrait having one, so the "core" sits on the baseline
+        val border = (actor as? Portrait)?.borderSize ?: 0f
+        // Scale to desired font dimensions - modifying the actor this way is OK as the decisions are
+        // the same each repetition, and size in the Group case or aspect ratio otherwise is preserved
+        if (actor is Group) {
```

Comment:
Split into subfunction, scaleAndPositionActor, with anything we can contain in that function